src/contexts/authContext.tsx

```diff
@@ -62,11 +62,20 @@ const useAuth: () => UseAuthHookReturnType = () => {
   useEffect(() => {
     if (token) {
       // fetch /refresh token
-      authApiService.refreshToken(token).then((data: OperationStatus<LoginDto>) => {
-        _setUser(data.payload?.user);
-        localStorage.setItem('token', token);
-        setIsLoading(false);
-      });
+      const refreshToken = async () => {
+        setIsLoading(true);
+        try {
+          authApiService.refreshToken(token).then((data: OperationStatus<LoginDto>) => {
+            _setUser(data.payload?.user);
+            localStorage.setItem('token', token);
+          });
+          setIsLoading(false);
```

Comment:
setting loading outside of the then statement will make the loader state false at the same time with starting the request, defeating the purpose of such loader. we should set the loading to false after the promise has been resolved. maybe on a 'finally' callback

Reply:
updated